test-data/unit/pythoneval-enum.test

```diff
@@ -132,3 +132,91 @@ class E(N, Enum):
 def f(x: E) -> None: pass
 
 f(E.X)
+
+[case testFunctionalEnumString]
+from enum import Enum, IntEnum
+E = Enum('E', 'foo bar')
+I = IntEnum('I', ' bar, baz ')
+reveal_type(E.foo)
+reveal_type(E.bar.value)
+reveal_type(I.bar)
+reveal_type(I.baz.value)
+[out]
+_program.py:4: error: Revealed type is '_testFunctionalEnumString.E'
+_program.py:5: error: Revealed type is 'Any'
+_program.py:6: error: Revealed type is '_testFunctionalEnumString.I'
+_program.py:7: error: Revealed type is 'builtins.int'
+
+[case testFunctionalEnumListOfStrings]
+from enum import Enum, IntEnum
+E = Enum('E', ('foo', 'bar'))
+F = IntEnum('F', ['bar', 'baz'])
+reveal_type(E.foo)
+reveal_type(F.baz)
+[out]
+_program.py:4: error: Revealed type is '_testFunctionalEnumListOfStrings.E'
+_program.py:5: error: Revealed type is '_testFunctionalEnumListOfStrings.F'
+
+[case testFunctionalEnumListOfPairs]
+from enum import Enum, IntEnum
+E = Enum('E', [('foo', 1), ['bar', 2]])
+F = IntEnum('F', (['bar', 1], ('baz', 2)))
+reveal_type(E.foo)
+reveal_type(F.baz)
+reveal_type(E.foo.value)
+reveal_type(F.bar.name)
+[out]
+_program.py:4: error: Revealed type is '_testFunctionalEnumListOfPairs.E'
+_program.py:5: error: Revealed type is '_testFunctionalEnumListOfPairs.F'
+_program.py:6: error: Revealed type is 'Any'
+_program.py:7: error: Revealed type is 'builtins.str'
+
+[case testFunctionalEnumDict]
+from enum import Enum, IntEnum
+E = Enum('E', {'foo': 1, 'bar': 2})
+F = IntEnum('F', {'bar': 1, 'baz': 2})
+reveal_type(E.foo)
+reveal_type(F.baz)
+reveal_type(E.foo.value)
+reveal_type(F.bar.name)
+[out]
+_program.py:4: error: Revealed type is '_testFunctionalEnumDict.E'
+_program.py:5: error: Revealed type is '_testFunctionalEnumDict.F'
+_program.py:6: error: Revealed type is 'Any'
+_program.py:7: error: Revealed type is 'builtins.str'
+
+[case testFunctionalEnumErrors]
```

Comment:
Add test cases:

* Incremental checking and `Enum(...)`
* Try to access missing `Enum` attribute.

What about adding a minimal stub for `enum` to `lib-stub` and moving these test cases away from `pythoneval-enum.test` to speed them up?